blazar/plugins/networks/network_plugin.py

```diff
@@ -795,6 +796,45 @@ def _filter_networks_by_properties(self, network_properties,
             return db_api.network_list()
 
 
+def remove_subnet_route_from_router(router_id, subnet_id, dry_run):
+    """Removes the route in the router pointing to the subnet
+
+    Args:
+        router_id (str): The router in which routes need to be checked and removed
+        subnet_id (str): the subnet to which routes should be removed
+    """
+    neutron_client = neutron.BlazarNeutronClient()
+    def is_route_pointing_to_subnet(route):
+        if 'gateway' in route:
+            next_hop_ip = ipaddress.ip_address(route['gateway'])
+        elif 'nexthop' in route:
+            next_hop_ip = ipaddress.ip_address(route['nexthop'])
+        subnet_cidr = ipaddress.ip_network(subnet['cidr'])
+        return next_hop_ip in subnet_cidr
+    try:
+        subnet = neutron_client.show_subnet(subnet_id).get('subnet', {})
+    except neutron_ex.NotFound as e:
```

Comment:
Maybe I am forgetting something, but is there a point to these try/except blocks if the exception is just reraised? Should there at least be a log statement here.

Reply:
Good catch! I removed them for some reason, The logs should be there